kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala

```diff
@@ -43,9 +44,14 @@ class SparkBatchProcessBuilder(
       buffer += cla
     }
 
+    // tag for YARN
     val batchJobTag = batchConf.get(TAG_KEY).map(_ + ",").getOrElse("") + batchId
+    var allConf = batchConf ++ Map(TAG_KEY -> batchJobTag) ++ sparkAppNameConf()
 
-    val allConf = batchConf ++ Map(TAG_KEY -> batchJobTag) ++ sparkAppNameConf()
+    // tag for K8S
+    conf.getOption("spark.kubernetes.driver.label." + LABEL_KYUUBI_UNIQUE_KEY).foreach(option => {
+      allConf = allConf ++ Map("spark.kubernetes.driver.label." + LABEL_KYUUBI_UNIQUE_KEY -> option)
+    })
```

Comment:
KyuubiApplicationManager.tagApplication

Reply:
I have noticed this, which needs to be linked with other files, and it has little to do with the goal of this PR, so it tends to be improved in the following PR.